ravenframework/DataObjects/DataSet.py

```diff
@@ -211,6 +231,13 @@ def addRealization(self, rlz):
     if indexMap is not None:
       # keep only those parts of the indexMap that correspond to variables we care about.
       indexMap = dict((key, val) for key, val in indexMap[0].items() if key in self.getVars()) # [0] because everything is nested in a list by now, it seems
+    #If the index is in autogenerate set, generate it automatically
+    if len(self._autogenerate) > 0:
+      for autoindex in self._autogenerate:
+        expectedLength = len(rlz[self._pivotParams[autoindex][0]])
+        #if it already exists and has correct length, don't add
+        if autoindex not in rlz or len(rlz[autoindex]) != expectedLength:
+          rlz[autoindex] = np.arange(len(rlz[self._pivotParams[autoindex][0]]))
```

Comment:
change len(rlz[self._pivotParams[autoindex][0]]) to expectedLength?

Reply:
I agree. fixed.